localstack/utils/cloudformation/template_deployer.py

```diff
@@ -518,6 +518,9 @@ def retrieve_resource_details(resource_id, resource_status, resources, stack_nam
     resource_props = resource.get('Properties')
     try:
         if resource_type == 'Lambda::Function':
+            resource_props['FunctionName'] = resource_props.get('FunctionName',
+                                                                'lambda-function-'.format(common.short_uid()))
```

Comment:
Not sure if the `format(..)` here would actually insert an identifier, or always return the same function name?

Nit: can we prefix the Lambda name with the CF stack name here:
```
resource_props['FunctionName'] = resource_props.get('FunctionName',
                                                                '%s-lambda-%s' % (stack_name, common.short_uid()))
```